docs/runtime_suite_applications/control-plane/20_configuration.mdx

```diff
@@ -0,0 +1,623 @@
+                }
+            }
+        }
+    },
+    "persistence": {
+        "type": "mongodb",
+        "configuration": {
+            "url": "mongodb://localhost:27017/fast-data?replicaSet=local"
+        },
+        "collection": {
+            "name": "runtime-state"
+        }
+    },
+    "settings": {
+        "server": {
+            "prefix": "/control-plane/",
+            "apis": {
+                "controllers": {
+                    "fast-data": "/my-awesome-controller"
+                }
+            },
+            "website": {
+                "url": "/__frontend/"
+            }
+        }
+    }
+}
+```
+
+</p>
+</details>
+
+<Tabs>
+
+<TabItem value="persistence" label="Persistence" default>
+
+### Persistence
+
+This section configures the layer of runtime state persistence where Control Plane stores the next desired state for each configured runtime.
+The client is also used to stream changes to the state channels of each given runtime.
+It must point to a `MongoDB` collection, available in a [`replicaSet` enabled database](https://www.mongodb.com/docs/manual/replication/). Available interfaces include:
+
+- `MongoDB` client
+
+Example:
+
+```json
+{
+    "type": "mongodb",
+    "configuration": {
+        "url": "mongodb://localhost:27017/fast-data?replicaSet=local"
+    },
+    "collection": {
+        "name": "runtime-state"
+    }
+}
+```
+
+</TabItem>
+
+<TabItem value="settings" label="Settings">
+
+### Settings
+
+The overall setup of Control Plane is tuned in the Settings object, allowing you to configure **HTTP controller** routes, including:
+
+- APIs
+  - Fast Data control endpoint
+  - Metrics
+  - Probes
+- Frontend/Website
+
+Example:
+
+```json
+{
+  "server": {
+    "prefix": "/control-plane/",
+    "website": {
+      "url": "/"
+    }
+  }
+}
+```
+
+</TabItem>
+
+<TabItem value="runtimes" label="Runtimes">
+
+### Runtimes
+
+A key-value object representing runtime instances controlled by this control-plane.
+Key is a unique name that corresponds with the persistence unique identifier for the current desired state of a given runtime.
+Runtimes contain communication channel from and to deployed microservices.
+For each deployed runtime, this section configures the communication channel that the Control Plane uses to dispatch the next desired state.
+Available interfaces are:
+
+- `Kafka` producer
+
+Example:
+
+```json
+{
+    "type": "kafka",
+    "configuration": {
+        "bootstrap.servers": "localhost:9092,localhost:9094",
+        "auto.offset.reset": "earliest"
+    },
+    "producer": {
+        "topic": "state-topic"
+    }
+}
+```
+
+</TabItem>
+
+</Tabs>
+
+### Compatibility Matrix
+
+Ensure compatibility with external services using the following matrix. Initially, only the persistence layer is required to be available at startup, as configured runtimes can be zero.
+
+| Control Plane | MongoDB | Kafka |
+|:-------------:|:-------:|:-----:|
+|      0.x      |  >=5    | >=3.5 |
```

Comment:
Should we set version 1.0.0 for Control Plane? Moreover, isn't Kafka version too restrictive?